deployer/pom.xml

```diff
@@ -16,23 +16,118 @@
+            <artifactId>brooklyn-test-support</artifactId>
+            <version>${brooklyn.version}</version>
+            <scope>test</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>org.apache.brooklyn</groupId>
+            <artifactId>brooklyn-core</artifactId>
+            <version>${brooklyn.version}</version>
+            <classifier>tests</classifier>
+            <scope>test</scope>
+        </dependency>
 
-  <groupId>eu.seaclouds-project</groupId>
-  <artifactId>deployer</artifactId>
-  <packaging>jar</packaging>
+        <dependency>
+            <groupId>org.apache.brooklyn</groupId>
+            <artifactId>brooklyn-launcher</artifactId>
+            <version>${brooklyn.version}</version>
+            <scope>test</scope>
+        </dependency>
 
-  <name>deployer</name>
-  <url>http://maven.apache.org</url>
+        <dependency>
+            <!-- this gives us flexible and easy-to-use logging; just edit logback-custom.xml! -->
+            <groupId>org.apache.brooklyn</groupId>
+            <artifactId>brooklyn-logback-xml</artifactId>
+            <version>${brooklyn.version}</version>
+            <!-- optional so that this project has logging; dependencies may redeclare or supply their own;
+                 provided so that it isn't put into the assembly (as it supplies its own explicit logback.xml);
+                 see Logging in the Brooklyn website/userguide for more info -->
+            <optional>true</optional>
+            <scope>provided</scope>
+        </dependency>
+    </dependencies>
 
-  <properties>
-    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-  </properties>
+    <build>
+        <plugins>
+            <plugin>
+                <artifactId>maven-assembly-plugin</artifactId>
+                <configuration>
+                    <descriptors>
+                        <descriptor>src/main/assembly/assembly.xml</descriptor>
+                    </descriptors>
+                    <appendAssemblyId>false</appendAssemblyId>
+                </configuration>
+                <executions>
+                    <execution>
+                        <id>make-assembly</id>
+                        <!-- this is used for inheritance merges -->
+                        <phase>package</phase>
+                        <!-- bind to the packaging phase -->
+                        <goals>
+                            <goal>single</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
 
+            <plugin>
+                <artifactId>maven-jar-plugin</artifactId>
+                <version>2.3.1</version>
+                <executions>
+                    <execution>
+                        <id>default-jar</id>
+                        <!-- put the default-jar in the none phase to skip it from being created -->
+                        <phase>none</phase>
```

Comment:
Currently we have added `deployer-0.8.0-SNAPSHOT.jar` as a  resource in `usage/installer/pom.xml`, as we can see following:

```
<resources>
    <resource>
        <directory>${basedir}/../../deployer/target</directory>
        <includes>
                <include>deployer-0.8.0-SNAPSHOT.jar</include>
        </includes>
    </resource>
        ...
</resources>
```
It allows to Brooklyn host ( which deploys `SeaCloudsPlatform` ) finds the `deployer-0.8.0-SNAPSHOT.jar` from the classpath and adds it to `BrooklynNode`. This jar ,that is generated by the assembly, includes all `cloudfoundry` and `mysql` dependencies. 

If we do not include the jar as a resource into `installer-0.8.0-SNAPSHOT.jar`, Brooklyn host will not find `deployer-0.8.0-SNAPSHOT.jar` from the classpath. 
We tried to add `deployer-0.8.0-SNAPSHOT.jar` to the classpath from maven, using a dependency, and the jar is added to the classpath but the jar is not found from the classpath.   because when Brooklyn receives a URL based on classpath, as following:
```
brooklynnode.classpath:
      - classpath://deployer-0.8.0-SNAPSHOT.jar
```
it tries to find this jar as a resource using `gerLocader().getResurce("deployer-0.8.0-SNAPSHOT.jar")`, however the `deployer-0.8.0-SNAPSHOT.jar` was not added as a resource, (it was added as maven dependency).

Then, if we do not want to build the custom `deployer-0.8.0-SNAPSHOT.jar` using the assembly, we should add the needed dependencies as resources too. It could be added to  `installer-0.8.0-SNAPSHOT.jar`, it was my first idea, however these dependencies (and the transitive dependencies) should be adde to `installer/pom.xml` from `.m2` repositories, something like following:
```
<resources>
    <resource>
        <directory>${user.home}.m2/repository/org/cloudfoundry/cloudfoundry-client-lib/1.1.2</directory>
         <includes>
             <include>cloudfoundry-client-lib-1.1.2</include>
         </includes>
    </resource>
</resources>
```
Next, this resource (jar dependencies) should to be added to `BrooklynNode` from the classpath.

I have developed this approach for checking if it works fine.